examples/text/plot_document_classification_20newsgroups.py

```diff
@@ -186,52 +270,48 @@ def benchmark(clf):
     print(metrics.confusion_matrix(y_test, pred))
 
     print()
-    clf_descr = str(clf).split("(")[0]
+    if custom_name:
+        clf_descr = str(custom_name)
+    else:
+        clf_descr = str(clf).split("(")[0]
     return clf_descr, score, train_time, test_time
 
 
 # %%
-# We now train and test the datasets with 15 different classification
+# We now train and test the datasets with 8 different classification
 # models and get performance results for each model.
-from sklearn.feature_selection import SelectFromModel
-from sklearn.linear_model import RidgeClassifier
-from sklearn.pipeline import Pipeline
+from sklearn.linear_model import LogisticRegression
 from sklearn.svm import LinearSVC
 from sklearn.linear_model import SGDClassifier
-from sklearn.linear_model import Perceptron
-from sklearn.linear_model import PassiveAggressiveClassifier
-from sklearn.naive_bayes import BernoulliNB, ComplementNB, MultinomialNB
+from sklearn.naive_bayes import ComplementNB
 from sklearn.neighbors import KNeighborsClassifier
 from sklearn.neighbors import NearestCentroid
 from sklearn.ensemble import RandomForestClassifier
 
 
 results = []
 for clf, name in (
-    (RidgeClassifier(tol=1e-2, solver="sag"), "Ridge Classifier"),
-    (Perceptron(max_iter=50), "Perceptron"),
-    (PassiveAggressiveClassifier(max_iter=50), "Passive-Aggressive"),
-    (KNeighborsClassifier(n_neighbors=10), "kNN"),
-    (RandomForestClassifier(), "Random forest"),
+    (LogisticRegression(C=5, max_iter=1000), "Logistic Regression"),
+    (RidgeClassifier(alpha=1.0, solver="sparse_cg"), "Ridge Classifier"),
+    (KNeighborsClassifier(n_neighbors=100), "kNN"),
+    (RandomForestClassifier(), "Random Forest"),
 ):
     print("=" * 80)
     print(name)
-    results.append(benchmark(clf))
-
-for penalty in ["l2", "l1"]:
-    print("=" * 80)
-    print("%s penalty" % penalty.upper())
-    # Train Liblinear model
-    results.append(benchmark(LinearSVC(penalty=penalty, dual=False, tol=1e-3)))
+    results.append(benchmark(clf, name))
 
-    # Train SGD model
-    results.append(benchmark(SGDClassifier(alpha=0.0001, max_iter=50, penalty=penalty)))
+# Train Liblinear model
+print("=" * 80)
+print("L2 penalty Linear SVC")
+results.append(benchmark(LinearSVC(C=0.1, dual=False, max_iter=1000)))
```

Comment:
I think all those models can be moved to the list in the for loop above.

We can put the descriptions such as "L2 penalty Linear SVC" as inline comment instead of print statements.